app.py

```diff
@@ -0,0 +1,437 @@
+        if attribute["key"] == attribute_name:
+            return attribute["value"]
+    return False
+
+
+def get_knack_record_id(citybase_data):
+    """
+    :param citybase_data: json from POST request
+    :return: str, knack record id or tuple(text, resp code), if error
+    """
+    try:
+        knack_record_id = get_custom_attribute(
+            citybase_data["data"]["custom_attributes"], "knack_record_id"
+        )
+    except KeyError:
+        return "Missing custom attributes", 400
+    if not knack_record_id:
+        return "Missing knack record id custom attribute", 400
+    return knack_record_id
+
+
+def get_knack_invoice(citybase_data):
+    """
+    :param citybase_data: json from POST request
+    :return: str, knack record id or tuple(text, resp code), if error
+    """
+    try:
+        knack_invoice = get_custom_attribute(
+            citybase_data["data"]["custom_attributes"], "invoice_number"
+        )
+    except KeyError:
+        return "Missing custom attributes", 400
+    if not knack_invoice:
+        return "Missing knack invoice number custom attribute", 400
+    return knack_invoice
+
+
+def get_knack_payload(environment, payment_status, today_date):
+    """
+    :param environment: "uat" or "prod" depending on which endpoint calls this function
+    :param payment_status: info from citybase payload
+    :param today_date: mm/dd/YYYY H:M datetime string
+    :return: json object to send along with PUT call to knack
+    """
+    return json.dumps(
+        {
+            field_maps[environment]["payment_status"]: payment_status_map[
+                payment_status
+            ],
+            field_maps[environment]["transaction_paid_date"]: today_date,
+        }
+    )
+
+
+def get_knack_refund_payload(
+    environment,
+    payment_status,
+    payment_amount,
+    knack_invoice,
+    today_date,
+    knack_record_id,
+):
+    """
+    :param environment: "uat" or "prod" depending on which endpoint calls this function
+    :param payment_status: info from citybase payload
+    :param payment_amount: string amount from citybase payload
+    :param knack_invoice: info from citybase payload
+    :param today_date: mm/dd/YYYY H:M datetime string
+    :param knack_record_id:
+    :return: json object to insert into transactions table in knack
+    """
+    refund_fields = {
+        "uat": {
+            "customer_name": "field_3334",
+            "event_name": "field_3348",
+            "type": "field_3337",
+            "banner_reservations_lpb": "field_3326",
+            "banner_reservations_ots": "field_3327",
+            "sub_description": "field_3349"
+        },
+        "prod": {
+            "customer_name": "field_3334",
+            "event_name": "field_3336",
+            "type": "field_3333",
+            "banner_reservations_lpb": "field_3328",
+            "banner_reservations_ots": "field_3329",
+            "sub_description": "field_3351"
+        },
+    }
```

Comment:
Would it be helpful to consolidate these field mappings into the global `field_maps` object on line 24? It looks like two of them are already represented there, but with different key names. I feel like I may be missing the reason that you've pulled them out down to define them in this function's scope, so please ignore this if I'm missing something.